### ISSUE TITLE
Janusgraph depends on dom4J which needs updating

### ISSUE THREAD

**Issue by planetf1** [odpi/egeria#1218]
Janusgraph will need updating once dom4J is upgraded to a good version

See https://github.com/JanusGraph/janusgraph/issues/1342

```
We pick up dom4J through:
[INFO] +- org.odpi.egeria:connector-configuration-factory:jar:1.1-SNAPSHOT:compile
[INFO] |  +- org.odpi.egeria:graph-repository-connector:jar:1.1-SNAPSHOT:compile
[INFO] |  |  +- org.janusgraph:janusgraph-core:jar:0.3.1:compile
[INFO] |  |  |  +- org.reflections:reflections:jar:0.9.9-RC1:compile
[INFO] |  |  |  |  +- org.javassist:javassist:jar:3.16.1-GA:compile
[INFO] |  |  |  |  \- dom4j:dom4j:jar:1.6.1:compile
```
cc: @grahamwallis

**Comment by grahamwallis** [comment]
@planetf1 - thanks for the heads up - I have subscribed to JanusGraph/janusgraph#1342.

Janusgraph latest release (0.3.2) and 0.4.0 pre-release both are still at reflections 0.9.9-RC1. I will monitor.

**Comment by planetf1** [comment]
Ref: CVE-2018-1000632

**Comment by planetf1** [comment]
dom4J leaks into many places via a similar route to lucene in https://github.com/odpi/egeria/issues/1106 - mostly through the graph repository connector

**Comment by grahamwallis** [comment]
If we could move Janus's dependency up to reflections 0.9.12, that would include dom4j 2.1.1 which would solve the CVE. I have asked the Janus team for any advice on how we could perform acceptance testing of a possible PR.

**Comment by grahamwallis** [comment]
As far as I can tell JG needs reflections 0.9.12 in order to overcome the CVE, but 0.9.12 has a bug so this is currently on hold. Keep open...

**Comment by grahamwallis** [comment]
JG is now using 1&1 reflections and elsewhere (e.g. in Egeria clients) we are using org.reflections 0.9.12, so I am going to close this; we can reopen if we get a CVE
